src/Factory.cc

```diff
@@ -69,23 +69,37 @@ std::vector<std::string> split(const std::string &_orig, char _delim)
 /////////////////////////////////////////////////
 /// \brief A factory class to generate protobuf messages at runtime based on
 /// their message descriptors. The location of the .desc files is expected
-/// via the IGN_DESCRIPTOR_PATH environment variable. This environment
+/// via the GZ_DESCRIPTOR_PATH environment variable. This environment
 /// variable expects paths to directories containing .desc files.
 /// Any file without the .desc extension will be ignored.
 class DynamicFactory
 {
   //////////////////////////////////////////////////
   /// \brief Constructor.
   /// The constructor will try to load all descriptors specified in the
-  /// IGN_DESCRIPTOR_PATH environment variable.
+  /// GZ_DESCRIPTOR_PATH environment variable.
   public: DynamicFactory()
   {
     // Try to get the list of paths from an environment variable.
-    const char *ignDescPaths = std::getenv("IGN_DESCRIPTOR_PATH");
+    const char *ignDescPaths = std::getenv("GZ_DESCRIPTOR_PATH");
     if (!ignDescPaths)
-      return;
+    {
+      // TODO(CH3): Deprecated. Remove on tock.
+      // Remember to still return !!
+      ignDescPaths = std::getenv("IGN_DESCRIPTOR_PATH");
+
+      if (!ignDescPaths)
+      {
+        return;
+      }
+      else
+      {
+        std::cout << "IGN_DESCRIPTOR_PATH is deprecated and will be removed! "
```

Comment:
I think `cerr` is more appropriate here. Yeah we need to resort to `iostream` for libraries that don't depend on `gz-common`. This makes it harder for users to suppress them at runtime, but we don't have a better option right now. In the future we may use an external logging library across the entire stack.

```suggestion
        std::cerr << "IGN_DESCRIPTOR_PATH is deprecated and will be removed! "
```